src/ibl_third_party_auth/views.py

```diff
@@ -15,3 +33,123 @@
+
+
+@method_decorator(csrf_exempt, name='dispatch')
+class DMTokenView(OAuthLibMixin, View):
+    server_class = OAuth2Server
+    validator_class = OAuth2Validator
+
+    oauthlib_backend_class = OAuthLibCore
+
+    @staticmethod
+    def parse_iso8601_to_utc(iso_string: str) -> datetime.datetime:
+        """
+        Parses an ISO 8601 formatted string into a timezone-aware UTC datetime object.
+    
+        - If the string contains timezone information (e.g., +02:00, -05:00, Z),
+          it's converted to UTC.
+        - If the string does *not* contain timezone information (naive datetime),
+          it's assumed to represent a UTC time, and UTC timezone info is attached.
+    
+        Args:
+            iso_string: The string formatted according to ISO 8601.
+    
+        Returns:
+            A timezone-aware datetime object representing the time in UTC.
+    
+        Raises:
+            ValueError: If the string is not a valid ISO 8601 format
+                        parsable by datetime.fromisoformat.
+        """
+        # Python < 3.11 doesn't handle 'Z' directly in fromisoformat
+        # We replace 'Z' with '+00:00' for compatibility
+        if iso_string.endswith('Z'):
+            iso_string = iso_string[:-1] + '+00:00'
+        try:
+            dt = datetime.datetime.fromisoformat(iso_string)
+            if dt.tzinfo is None or dt.tzinfo.utcoffset(dt) is None:
+                dt_utc = dt.replace(tzinfo=datetime.timezone.utc)
+            else:
+                dt_utc = dt.astimezone(datetime.timezone.utc)
+            return dt_utc
+    
+        except ValueError as e:
+            raise ValueError(f"Invalid ISO 8601 string format: '{iso_string}'") from e
+
+    def post(self, request, *args, **kwargs):
+        url, headers, body, status = self.create_token_response(request)
+        if status == 200:
+            access_token = json.loads(body).get("access_token")
+            token = get_access_token_model().objects.get(token=access_token)
+            try:
+                log.info(token.user.email)
+                log.info(token.user.username)
+                log.info(token.user.id)
+                response = manager_api_request(
+                    "POST", MANAGER_TOKEN_ENDPOINT_PATH, data={"email": token.user.email}
+                )
+
+                try:
+                    data = response.json()
+                except ValueError:
+                    if response.ok:
+                        # Only log when the response is expected to be valid
+                        log.exception(
+                            "Non-JSON token proxy response: %s %s",
+                            response.status_code,
+                            response.text,
+                        )
+
+                    data = None
+            except Exception:
+                log.exception("Token proxy request error")
+                data = None
+            if data and not data.get("token"):
+                data = None
```

Comment:
If there's data but no token, then we probably want to log an error here? This feels like it would be something we would need to investigate so we would want to log relevant information so we could dig in.